assets/src/edit-story/components/panels/layer/panel.js

```diff
@@ -23,13 +23,19 @@ import { __ } from '@wordpress/i18n';
  * Internal dependencies
  */
 import { Panel, PanelTitle, PanelContent } from '../panel';
-import { DEFAULT_LAYERS_VISIBLE, LAYER_HEIGHT } from './constants';
+import { LAYER_HEIGHT, DEFAULT_LAYERS_VISIBLE } from './constants';
 import LayerList from './layerList';
 import LayerProvider from './provider';
+import useLayers from './useLayers';
 
 function LayerPanel() {
+  const layers = useLayers();
+
   return (
-    <Panel name="layers" initialHeight={DEFAULT_LAYERS_VISIBLE * LAYER_HEIGHT}>
+    <Panel
+      name="layers"
+      initialHeight={(layers?.length || DEFAULT_LAYERS_VISIBLE) * LAYER_HEIGHT}
```

Comment:
Is this aligned with @samitron7? The design originally called for always 6 initial rows. We have #227 too, by the way, which is then fixed by this issue?

Reply:
Tagging @samitron7 here but from our discussion it seemed like she wanted it to vary by number of items more than page size